### ISSUE TITLE
[Bug]: Firewall not removed

### ISSUE THREAD

**Issue by MicheleSica** [hetznercloud/terraform-provider-hcloud#477]
### What happened?

We have created a bunch of servers without any issue with `terraform apply`
At the destroy step: `terraform destroy` 
the firewall resource named firewall_1 has not been removed despite the message: `hcloud_firewall.firewall_1: Destruction complete after 0s`

### What did you expect to happen?

We did expect the firewall to be destroyed

### Please provide a minimal working 

```
provider "hcloud" {
  token = ""

}

terraform {
  required_providers {
    hcloud = {
      source = "hetznercloud/hcloud"

      version = "1.32.0"
    }
  }

}

resource "hcloud_server" "master" {

  name         = "master-001"
  image        = var.os_image
  server_type  = var.instance_type
  ssh_keys     = [hcloud_ssh_key.this.name]
  datacenter   = var.datacenter
  firewall_ids = [hcloud_firewall.firewall_1.id]
  labels = {
    server = "master"
  }

}

variable "datacenter" {
  default = "nbg1-dc3"
}

variable "os_image" {
  default = "ubuntu-20.04"
}

variable "instance_type" {
  default = "cpx11"
}

resource "random_string" "random" {
  length           = 16
  special          = true
  override_special = "/@£$"
}

resource "hcloud_network" "network" {
  name     = join("-", ["hetzner-kubeadm", random_string.random.id])
  ip_range = "10.0.0.0/16"
}

resource "hcloud_network_subnet" "subnet_pub" {
  network_id   = hcloud_network.network.id
  type         = "cloud"
  network_zone = "eu-central"
  ip_range     = "10.0.1.0/24"
}

resource "hcloud_server_network" "master_network" {
  server_id  = hcloud_server.master.id
  network_id = hcloud_network.network.id
}

resource "hcloud_ssh_key" "this" {
  name       = ""
  public_key = file("~/.ssh/id_rsa.pub")
}

resource "hcloud_firewall" "firewall_1" {
  name = "firewall_1"
  rule {
    direction = "in"
    protocol  = "icmp"
    source_ips = [
      "0.0.0.0/0",
      "::/0"
    ]
  }
}
```

**Comment by AndreiPaulau** [comment]
+1
I see in terraform logs:
`[0m[1mhcloud_firewall.app-firewall: Destroying... [id=199431][0m[0m
[0m[1mhcloud_firewall.app-firewall: Destruction complete after 0s[0m`

But in https://console.hetzner.cloud/ there is no audit event related to the firewall (the rest resources were deleted successfully)

I have 4 different project where we use the same snippet of tf firewall rule creation.
3 project works well, the last one - not.

hetznercloud/hcloud v1.32.0
and yes, looks like it is related to 9e08f76 ,since it's started 4-5 days ago according CI logs

**Comment by fhofherr** [comment]
Hi @MicheleSica,

this is fixed with release 1.32.1 (#478). I'll close this issue for now. Feel free to re-open it if it does not work for you.
